modules/tests/src/test/scala/RuntimeTest.scala

```diff
@@ -0,0 +1,52 @@
+package tests.runtime
+
+import langoustine.lsp.runtime.*
+import scala.language.strictEquality
+
+object RuntimeTest extends weaver.FunSuite:
+  test("Opt[A] is null at runtime"):
+      assert.all(
+        Opt.empty == null,
+        Opt(null) == null,
+        Opt("yo") != null
+      )
+
+  test("Opt[A] is A at runtime"):
+      assert.all(
+        Opt(25).asInstanceOf[Int] == 25,
+        Opt("hello").asInstanceOf[String] == "hello"
+      )
+
+  test("Opt[A] <-> Option[A] interop"):
+      assert.all(
+        Opt("howdy").toOption.contains("howdy"),
+        Opt.empty.toOption.isEmpty,
+        Opt.fromOption(Some("yesss")) == Opt("yesss"),
+        Opt.fromOption(None) == Opt.empty
+      )
+
+  test("Opt[A] can be pattern matched on"):
+      Opt("yo") match
+        case Opt(str)  => expect(str == "yo")
+        case Opt.empty => failure("expected value to be present")
+
+  test("Empty Opt[A] can be pattern matched on"):
+      Opt.empty match
+        case Opt(str)  => failure("expected value to be present")
+        case Opt.empty => success
+
+  test("Opt[A]: Checks under strict equality"):
+      val patternMatchSome = Opt("yo") match
+        case Opt(str)  => expect(str == "yo")
+        case Opt.empty => failure("expected value to be present")
+
+      val patterMatchNone = Opt.empty match
+        case Opt(str)  => failure("expected value to be present")
+        case Opt.empty => success
+
+      val nullChecks =
+        assert.all(Opt.empty == null, Opt(null) == null, Opt("yo") != null)
+
+      patternMatchSome && patterMatchNone && nullChecks
+
+end RuntimeTest
```

Comment:
@kubukoz I think those tests show improved usability, let me know what you think

The `Opt[Opt[..]]` pattern is a problem, as is with all non-allocating Option-like wrappers.

General recommendation is to not let `Opt` leak out of any code that doesn't deal with the immediate payloads, and hence we provide an Option interop.

In the future Opt might go away - it exists for two purposes:
1. Faithfully match the LSP spec that is full of nulls, but provide some typesafety
2. Allow to reconfigure upickle's codec which for Option is unusable in the LSP scenario

Reply:
yeah, that looks good